### PR TITLE
[Fix] Improve provider's status check flow

### PR DESCRIPTION
**Pull request by fpezzola**
# Name of the feature/issue
Improve provider's status check flow

## Description
The extensions is overflowing the nodes when the provider's is down. With this PR we:
- Check provider's status when it there is an active subscription to the wallet and it is unlocked. Otherwise, the check is turned off.
- Await for the `ehters.provider` response (max 10 seconds) before enqueue a new provider's check.
- Check that the network hasn't changed when getting `ethers.providers` response.

### CODE REVIEW

File: packages/background/src/controllers/NetworkController.ts [hunk 1]
```diff
@@ -73,12 +73,14 @@ export default class NetworkController extends BaseController<NetworkControllerS
...
+        this.provider.on('error', (err) => {
+            return this._updateProviderNetworkStatus({ err });
+        });
 
         // Periodical checks when provider is down
-        setInterval(() => {
+        setInterval(async () => {
             if (!this.getState().isProviderNetworkOnline) {
-                this._updateProviderNetworkStatus();
+                await this._updateProviderNetworkStatus({ timeout: 3000 });
```
Comment (julianariel) [inline packages/background/src/controllers/NetworkController.ts:83]: shouldn't we move this timeout value to a constant? could be potentially included in the remote config repo along with other network settings, so maybe it makes sense to have it on a constant so we don't forget about it

**Review by julianariel — APPROVED** [review]
LGTM! left small comment

File: packages/background/src/controllers/NetworkController.ts (hunk 1 above)
Comment (fpezzola) [reply packages/background/src/controllers/NetworkController.ts]: This code has changed due to some features lacks I found in the `ethers.provider` implementation. 

We use the `poll` method to query the provider's status where we specify a 10seconds of timeout. However, this a virtual timeout since the library doesn't actually cancel the request, it only rejects the promise.
That being said, In 30 seconds of failure scenario we: 
- Check the provider's status => 10 seconds (1st request)
- Get the response and sleep => 5 seconds = 15 seconds
- Check the provider's status => 10 seconds (2nd request)  = 25 seconds 
- Get the response and sleep => 5 seconds = 30 seconds
- Check the provider's status => 10 seconds (3rd request) = 40 seconds
- Get the response and sleep => 5 seconds = 45 seconds
- Check the provider's status => 10 seconds (4th request) = 55 seconds
- Get the response and sleep => 5 seconds = 1 minute => 1st request is timeouted by the browser. 

In summary, in 1 minute we will have a max of 4 active request vs 20 request we used to have.
